File: Easy/1332. Remove Palindromic Subsequences/removePalindromeSub.java
class Solution {
    public int removePalindromeSub(String s) {
        if(s.length() == 0){
            return 0;
        }
        String s2 = "";
        for(int i = s.length()-1; i >= 0; i--){
            s2 += s.charAt(i);
        }
        if(s.equals(s2)){
            return 1;
        }
        return 2;
    }
}

File: Easy/1332. Remove Palindromic Subsequences/removePalindromeSub.cpp
/*
	如果本身就是空字串時就不用進行移除所以回傳0
	如果本身就是一個回文字串就直接移除回傳1
	其他的話 由於題目說字串都是由a和b構成的
	所以我們可以一次先全部移除a 這樣也是一種回文
	再一次全部移除b 所以回傳2
	Time complexity: O(n)
	Space complexity: O(n) / O(1)
*/
class Solution {
public:
    int removePalindromeSub(string s) {
        if(s.empty()){
            return 0;
        }
        if(s == string(s.rbegin(), s.rend())){
            return 1;
        }
        return 2;
    }
};